File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    hashMap := make(map[int]int)
    answer := make([]int,2)
    for i,v := range nums {
        diff := target - v
        _, ok := hashMap[diff]
        if ok {
            answer = []int{hashMap[diff],i}
            break
        } else {
            hashMap[v] = i
        }
    }
    return answer
}

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hashMap = new HashMap<>();
        int[] answer = new int[2];
        for (int i = 0; i < nums.length; i++) {
            int difference = target - nums[i];
            if (hashMap.containsKey(difference)) {
                answer[0] = hashMap.get(difference);
                answer[1] = i;
            }
            hashMap.put(nums[i], i);
        }
        return answer;
    }
}